### PR TITLE
refs issue:11706 Resolving the Alias value of Alias type node

### PR DESCRIPTION
**Pull request by arankeparth**
**Changes:**

1.  If the node is of `Alias` kind, we need to preserve the current path and process the Alias value separately.
2. By doing this, the `!overwrite ` tag won't be ignored.

**Testing:**

1. Created setup that mentioned on this issue: https://github.com/docker/compose/issues/11706
2. Results:

containers are spawned successfully:

Here, we can see in the configs section for `y` that there is only one entry for `configs`

### CODE REVIEW

**Comment by arankeparth** [comment]
Please review somebody

**Comment by arankeparth** [comment]
@ndeloof please review.

File: loader/reset.go
Comment (ndeloof) [inline loader/reset.go:46]: use path.Parts()

File: loader/reset.go
Comment (ndeloof) [inline loader/reset.go:50]: path = path.Next(el)

File: loader/reset.go
Comment (ndeloof) [inline loader/reset.go:45]: isn't this just equivalent to `path = strings.Replace(path, ".<<", "")` ?

**Review by ndeloof — COMMENTED** [review]
Can you please add a test case to demonstrate this actually fixes the merge issue?

File: loader/reset.go
Comment (arankeparth) [reply loader/reset.go]: Yes, I have made this change.

**Review by ndeloof — APPROVED** [review]

**Comment by arankeparth** [comment]
@ndeloof how to get this merged?

**Comment by ndeloof** [comment]
@arankeparth let's wait for a second review, then I think we can merge

**Comment by arankeparth** [comment]
@glours please review.

**Comment by arankeparth** [comment]
@milas can you please review this?

**Comment by ndeloof** [comment]
@arankeparth please don't push pressure on the team, will be reviewed once we get bandwidth to review

**Comment by arankeparth** [comment]
> @arankeparth please don't push pressure on the team, will be reviewed once we get bandwidth to review

Sorry for that.

**Review by glours — APPROVED** [review]

**Pull request merged**
